src/dftbp/solvation/solvparser.F90

```diff
@@ -447,7 +447,7 @@ subroutine readCM5(node, input, geo)
     end if
     input%atomicRad(:) = input%atomicRad * conv
 
-    call getChildValue(node, "Cutoff", input%rCutoff, 30.0_dp, &
```

Comment:
Is there a reason for removing the default of 30 Bohr?

Reply:
I didn't remove it, I moved the default definition from here within the parser to the definition of `TCM5Input` in `cm5.f90`, since this allows to construct the cm5 input with defaults without parsing a file and it made sense to me in a general sense, but I saw that you have many defaults in the parser so I don't know how you would want to handle that.

Alternatively one would need to duplicate the defaults in the cm5 construction, which I tried to avoid, since it is an invitation for diverging code in the future. 

Previously the cm5 model could not be constructed without parsing something (due to these defaults). I implemented that so the API call does not fail, if the input is missing the CM5 keyword

Reply:
I'd like to add additionally, that IMO this is not valid Fortran. :wink:  You are passing the same actual argument for two different dummy arguments, one them being `intent(out)`, the other being `intent(in)`. The compiler could in theory reset/nullify the `intent(out)` at entry at the subroutine, but that would change the value of your `intent(in)` argument, where you passed the same variable.
If we really wanted this (in order to ensure, that you use the default value stored in the default initialization of the derived type) you would have to do:
```
call getChildValue(node, "Cutoff", input%rCutoff, (input%rCutoff), &
```
which would create an expression (a copy) for the `intent(in)` argument.
But I personally would also opt for the more explicit original initialization here, I think, that is less error-prone.

Reply:
Ah tricky Fortran being tricky^^

Thanks for pointing it out